### ISSUE TITLE
CRL lastUpdate check handles sec == 0 case wrong.

### ISSUE THREAD

**Issue by davidben** [openssl/openssl#1352]
In 626aa24849be549b7ef4f049d8427989940c8a37, the lastUpdate comparison uses:

```
if (day < 0 || sec <= 0)
```

I do not believe the `sec <= 0` clause works. If `sec < 0`, we are guaranteed that `day <= 0` and do not need to check `day`, but this is not the case for `sec == 0`. In that case, we may have `day > 0` and this check will incorrectly act as if `crl` is earlier than `best_crl` when it is actually later. It should be:

```
if (day < 0 || (day == 0 && sec <= 0))
```

which I believe is equivalent to just:

```
/* ASN1_TIME_diff never returns inconsistent signs for |day| and |sec|. */
if (day <= 0 && sec <= 0)
```

This will need to be fixed on the 1.0.2 branch too.

PS: This change and the original probably should have some tests.
PPS: This is sufficiently hairy of a check that an `ASN1_TIME_cmp` may be worthwhile.

**Comment by davidben** [comment]
I think another option would be
```
if (day < 0 || sec < 0)
```
This would be logically `crl.lastUpdate < best_crl.lastUpdate` rather than `crl.lastUpdate <= best_crl.lastUpdate`.

(That's interesting that `&&` switches to `||` when you go from `<=` to `<`. More reason to hide this behind an `ASN1_TIME_cmp`.)

**Comment by mattcaswell** [comment]
Steve fixed this in e032117db2519. Closing.
